client/js/jsonedit.js

```diff
@@ -1475,16 +1483,59 @@ function jeSetAndSelect(replaceBy, insideString) {
   else
     var jsonString = JSON.stringify(jeStateNow, null, '  ');
   const startIndex = jsonString.indexOf(insideString ? '###SELECT ME###' : '"###SELECT ME###"');
-  let length = jsonString.length-15-(insideString ? 0 : 2);
-
+  const replaceByString = JSON.stringify(replaceBy);
   if(insideString)
-    jsonString = jsonString.replace(/###SELECT ME###/, JSON.stringify(replaceBy).substr(1, JSON.stringify(replaceBy).length-2));
+    jsonString = jsonString.replace(/###SELECT ME###/, replaceByString.substr(1, replaceByString.length-2));
   else
-    jsonString = jsonString.replace(/"###SELECT ME###"/, JSON.stringify(replaceBy));
+    jsonString = jsonString.replace(/"###SELECT ME###"/, replaceByString);
 
   jeSet(jsonString);
-  const quote = typeof replaceBy == 'string' && !insideString ? 1 : 0;
-  jeSelect(startIndex + quote, startIndex+jsonString.length-length - quote, true);
+
+  let bracket = 0;
+  let wavybracket = 0;
+  let begin = 0;
+  let end = 0;
+  if(replaceBy !== undefined){ 
+    const quoteArray = getAllIndexes(replaceByString,'\"');
+    if(quoteArray.length > 2) { // adjusts selection area to inside last set of quotes
+      begin = quoteArray[2] + 1;
+      end = quoteArray[3];
+    } else if (quoteArray.length > 0) {
+      begin = quoteArray[0] + 1;
+      end = quoteArray[1];
+    } else { // adjusts selection area to inside brackets
+      bracket = replaceByString.lastIndexOf('[') + 1;
+      wavybracket = replaceByString.lastIndexOf('{')+ 1;
+      begin = Math.max(bracket, wavybracket, 0);
+      bracket = replaceByString.lastIndexOf(']') > 0 ? replaceByString.lastIndexOf(']') > - 1 : replaceByString.length;
+      wavybracket = replaceByString.lastIndexOf('}') > 0 ? replaceByString.lastIndexOf('}') > - 1 : replaceByString.length;
+      end = Math.min(bracket, wavybracket);
+    }
+
+    // adjusts selection area for formatting of arrays and objects
+    if((replaceByString.indexOf('[') > -1 && (replaceByString.indexOf('\"') > -1 || replaceByString.indexOf('{') > -1)) || replaceByString.indexOf('\"') > -1 && replaceByString.indexOf('{') > -1) {
+      begin = begin + 6;
+      end = end + 6;
+      if(replaceByString.indexOf('{') > -1 && replaceByString.indexOf('\"') > -1 && replaceByString.indexOf('[') > -1) {
+        begin = begin + 7;
+        end = end + 7;
+      }
+    }
+    // variables with dollar signs and dynamic expressions
+    if(replaceByString.includes(String.fromCharCode(36))) {
+      begin = 3
+      end = 3
+    }
+    if(replaceByString.includes('increment')) {
+      begin = 14
+      end = 26
```

Comment:
Personally, I'd highlight the entire statement after `var `; it seems unlikely that any of that statement would be used. In fact, I don't really understand why there are two `var` variants offered as options in the first place; in 99% of the cases the user will simply delete everything after the `var `, I would think. But that's another question.

Reply:
Maybe that is a better idea, or maybe we should just add a 3rd expression of `"var "` and place the cursor before the last quote.  It appears it would indent correctly (though not color as a dynamic expression).

Perhaps ArnoldSmith86 though the increment and random were used enough to justify their specific inclusion - I don't really know, and am okay with either way.